src/main/java/com/github/kornilova_l/flamegraph/plugin/server/converters/file/jmc/FlightRecorderCompressedConverter.java

```diff
@@ -0,0 +1,87 @@
+package com.github.kornilova_l.flamegraph.plugin.server.converters.file.jmc;
+
+import com.github.kornilova_l.flamegraph.plugin.server.converters.file.CFlamegraph;
+import com.github.kornilova_l.flamegraph.plugin.server.converters.file.CFlamegraphLine;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+class FlightRecorderCompressedConverter extends FlightRecorderConverter {
+    private CFlamegraph cFlamegraph;
+
+    public FlightRecorderCompressedConverter(File file) {
+        super(file);
+
+        parseAsCFlamegraph();
+    }
+
+    CFlamegraph getCFlamegraph() {
+        return cFlamegraph;
+    }
+
+    private void parseAsCFlamegraph() {
+        ArrayList<CFlamegraphLine> cFlamegraphLines = new ArrayList<>();
+        HashMap<String, Integer> classNames = new HashMap<>();
+        HashMap<String, Integer> methodNames = new HashMap<>();
+        HashMap<String, Integer> descriptions = new HashMap<>();
+
+        Map<String, Integer> stacks = this.getStacks();
+        Set<String> stackFrames = stacks.keySet();
+
+        for(String frame : stackFrames) {
+            Integer width = stacks.get(frame);
+
+            String[] calls = frame.split(";");
+            Integer depth = calls.length;
+            String topFrame = calls[calls.length - 1]; //TODO: Add a check on length before this access
+            String className = getClassName(topFrame);
+            String methodName = getMethodName(topFrame);
+
+            CFlamegraphLine line = new CFlamegraphLine(
+                    getId(classNames, className),
+                    getId(methodNames, methodName),
+                    null,
+                    width,
+                    depth);
+            cFlamegraphLines.add(line);
+        }
+
+        this.cFlamegraph = new CFlamegraph(cFlamegraphLines, toArray(classNames), toArray(methodNames), toArray(descriptions));
+    }
+
+    private Integer getId(HashMap<String, Integer> map, String name) {
```

Comment:
Move `getId` and `toArray` functions from `....file.yourkit.csv.Converter` to `ProfilerToCompressedFlamegraphConverter` file and call it from both places

Reply:
Our inheritance needed some work to support this. `....file.yourkit.csv.Converter` doesn't extend or implement any other classes, but it handles the hard work in `YourkitCsvToCFlamegraphConverter.convert`. I created a new Abstract class (`AbstractConverter`) so both `FlightRecorderCompressedConverter` and `....file.yourkit.csv.Converter` could extend it.